protocol-designer/src/ui/steps/actions/thunks/index.ts

```diff
@@ -257,3 +259,68 @@ export const saveSetTempFormWithAddedPauseUntilTemp: () => ThunkAction<any> = ()
     assert(false, 'could not auto-save pause form, getUnsavedForm returned')
   }
 }
+
+export const saveHeaterShakerFormWithAddedPauseUntilTemp: () => ThunkAction<any> = () => (
+  dispatch,
+  getState
+) => {
+  const initialState = getState()
+  const unsavedSetHeaterShakerTemperatureForm = getUnsavedForm(initialState)
+  const isPristineSetHeaterShakerTempForm = getUnsavedFormIsPristineHeaterShakerForm(
+    initialState
+  )
+
+  if (!unsavedSetHeaterShakerTemperatureForm) {
+    assert(
+      false,
+      'Tried to saveSetHeaterShakerTempFormWithAddedPauseUntilTemp with falsey unsavedForm. This should never be able to happen.'
+    )
+    return
+  }
+
+  const { id } = unsavedSetHeaterShakerTemperatureForm
+
+  if (!isPristineSetHeaterShakerTempForm) {
+    assert(
+      false,
+      `tried to saveSetHeaterShakerTempFormWithAddedPauseUntilTemp but form ${id} is not a pristine set heater shaker temp form`
+    )
+    return
+  }
+
+  const temperature =
+    unsavedSetHeaterShakerTemperatureForm?.targetHeaterShakerTemperature
+
+  assert(
+    temperature != null && temperature !== '',
+    `tried to auto-add a pause until temp, but targetHeaterShakerTemperature is missing: ${temperature}`
+  )
+  dispatch(_saveStepForm(unsavedSetHeaterShakerTemperatureForm))
+  dispatch(
+    addStep({
+      stepType: 'pause',
+      robotStateTimeline: fileDataSelectors.getRobotStateTimeline(getState()),
+    })
+  )
+  dispatch(
+    changeFormInput({
+      update: {
+        pauseAction: PAUSE_UNTIL_TEMP,
+      },
+    })
+  )
```

Comment:
right after we dispatch this `PAUSE_UNTIL_TEMP` form change action, we also need change the module id in pause form to the h-s module id:
```suggestion
  )
    const heaterShakerModuleId = unsavedSetHeaterShakerTemperatureForm.moduleId
    dispatch(
      changeFormInput({
        update: {
          moduleId: heaterShakerModuleId,
        },
      })
    )
```